server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -284,6 +284,15 @@ public Map<String, CompatibilityVersions> compatibilityVersions() {
         return this.compatibilityVersions;
     }
 
+    public boolean hasMixedSystemIndexVersions() {
+        for (var entry : this.compatibilityVersions.entrySet()) {
+            if (false == entry.getValue().systemIndexMappingsVersion().equals(minVersions.systemIndexMappingsVersion())) {
```

Comment:
The equality check is the wrong way round. Also you could just use streams here?

`compatiblityVersions.values().stream().anyMatch(e -> e.systemIndexMappingsVersion().equals(minVersions...) == false)`